### PR TITLE
[17.0][ADD] product_category_pricelist_button

### PR DESCRIPTION
**Pull request by Anxo82**

### CODE REVIEW

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks

File: product_category_pricelist_button/models/product_category.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    pricelist_count = fields.Integer(compute='_compute_pricelist_count',
+        help="The number of pricelists associated with products under this category."
+    )
+
+    def _compute_pricelist_count(self):
+        for category in self:
+            category.pricelist_count = self.env['product.pricelist.item'].search_count(
+                [('categ_id', 'in', category.ids)]
+            )
```
Comment (dalonsod) [inline product_category_pricelist_button/models/product_category.py]: We'll try to obtain this count for every category in a more efficient way. Please take a look at this:

https://github.com/OCA/purchase-workflow/blob/8c0e713baac1d9a16356316a2fbfecf048ce46ec/purchase_order_type_dashboard/models/purchase_order_type.py#L32-L58

(purchase order count for certain purchase order types). This example is more complex than we require, please try to adapt it here (if find a simpler example I'll provide it anyway)

File: product_category_pricelist_button/views/product_category_views.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<odoo>
+    <record id="view_product_category_form_inherit" model="ir.ui.view">
```
Comment (dalonsod) [inline product_category_pricelist_button/views/product_category_views.xml:2]: Once items count is refactored please add that field to tree view

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments for suggested enhacements

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks

**Comment by dalonsod** [comment]
@lmiguens-solvos form view, as expected:

imagen


...but tree view is terribly simple:

imagen

(@Anxo82 we should tell them as "Pricelist Items", not "Pricelists")

Would you need some extra features in this view, like highlighting those categories that have one ore more pricelist items, or some filters "With pricelist items" and "Without pricelist items"?

**Comment by lmiguens-solvos** [comment]
It's okay, at the moment no more extra features are necessary. 
I also agree with the change ("Pricelist Items" for "Pricelists")) that you indicate.
And I'm waiting to see how that pricelist item tree view looks...

**Comment by Anxo82** [comment]
@dalonsod @lmiguens-solvos Review please.
Thanks

File: product_category_pricelist_button/models/product_category.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+from odoo import models, fields
+
+class ProductCategory(models.Model):
+    _inherit = 'product.category'
+
+    pricelist_count = fields.Integer(compute='_compute_pricelist_count', store=True,
+        help="The number of pricelists associated with products under this category."
+    )
+
+    def _compute_pricelist_count(self):
```
Comment (dalonsod) [inline product_category_pricelist_button/models/product_category.py:18]: Once `pricelist_count` is now stored, an `@api.depends` clause should be added, could you make the required modifications?

**Comment by Anxo82** [comment]
@dalonsod Review please.
Thanks.

File: product_category_pricelist_button/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+{
+    "name": "Product category pricelist button",
+    "summary": """
+        Adds a smart button to product categories linking to associated pricelists.
```
Comment (dalonsod) [inline product_category_pricelist_button/__manifest__.py]: Pending items reference:
```suggestion
        Adds a smart button to product categories linking to associated pricelist items.
```

File: product_category_pricelist_button/readme/DESCRIPTION.md [hunk 5]
```diff
@@ -0,0 +1 @@
+Adds a smart button to product categories linking to associated pricelists.
```
Comment (dalonsod) [inline product_category_pricelist_button/readme/DESCRIPTION.md]: Here too:
```suggestion
Adds a smart button to product categories linking to associated pricelist items.
```

**Review by dalonsod — APPROVED** [review]
:+1: 

Please make those pending descriptions issues, generate `README` stuff, make squashes, and I'll merge it

**Comment by Anxo82** [comment]
@dalonsod Done, review please.
Thanks

**Pull request merged**